src/lib/routes/admin-api/project/variants.ts

```diff
@@ -193,6 +223,62 @@ export default class VariantsController extends Controller {
         });
     }
 
+    async pushVariantsToEnvironments(
+        req: IAuthRequest<FeatureEnvironmentsParams, any, IVariant[], any>,
+        res: Response<FeatureVariantsSchema>,
+    ): Promise<void> {
+        const { projectId, featureName } = req.params;
+        const environments = req.query.environments?.split(','); // TODO can we use query-string parser?
+        const userName = extractUsername(req);
+
+        if (environments === undefined || environments.length === 0) {
+            throw new BadDataError('No environments provided');
+        }
+
+        await this.checkAccess(
+            req.user,
+            projectId,
+            environments,
+            UPDATE_FEATURE_ENVIRONMENT_VARIANTS,
+        );
+
+        const variants: IVariant[] = req.body;
+        await this.featureService.setVariantsOnEnvs(
+            projectId,
+            featureName,
+            environments,
+            variants,
+            userName,
+        );
+        res.status(200).json({
+            version: 1,
+            variants: variants,
```

Comment:
Worth noticing that here we're just echoing the variants received as parameter we could potentially collect the responses from each environment and return an object with variants per each environment or we could also just return a 202 without body...